sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java

```diff
@@ -108,6 +108,8 @@ public class DataLakePathAsyncClient {
 
     private final AzureSasCredential sasToken;
 
+    private final boolean isTokenCredentialAuthenticated;
```

Comment:
Boolean was added so that we can determine whether or not user is authenticated using token credential for paginated delete. Check whether the addition of this boolean field is necessary though, and if instead we can perform paginated delete without checking for token credential and gracefully fail.